packages/components/tabs/src/tab-nav.ts

```diff
@@ -148,25 +149,36 @@ export default defineComponent({
       const currentOffset = navOffset.value
       let newOffset = currentOffset
 
+      let startDiff, endDiff, centerDiff
       if (isHorizontal) {
-        if (activeTabBounding.left < navScrollBounding.left) {
-          newOffset =
-            currentOffset - (navScrollBounding.left - activeTabBounding.left)
-        }
-        if (activeTabBounding.right > navScrollBounding.right) {
-          newOffset =
-            currentOffset + activeTabBounding.right - navScrollBounding.right
-        }
+        startDiff = activeTabBounding.left - navScrollBounding.left
+        endDiff = activeTabBounding.right - navScrollBounding.right
+        centerDiff =
+          startDiff - navScrollBounding.width / 2 + activeTabBounding.width / 2
       } else {
-        if (activeTabBounding.top < navScrollBounding.top) {
-          newOffset =
-            currentOffset - (navScrollBounding.top - activeTabBounding.top)
-        }
-        if (activeTabBounding.bottom > navScrollBounding.bottom) {
-          newOffset =
-            currentOffset +
-            (activeTabBounding.bottom - navScrollBounding.bottom)
-        }
+        startDiff = activeTabBounding.top - navScrollBounding.top
+        endDiff = activeTabBounding.bottom - navScrollBounding.bottom
+        centerDiff =
+          startDiff -
+          navScrollBounding.height / 2 +
+          activeTabBounding.height / 2
+      }
+
+      const scrollPosition =
+        props.scrollPosition === 'auto'
+          ? startDiff < 0
+            ? 'start'
+            : endDiff > 0
+            ? 'end'
+            : null
+          : props.scrollPosition
```

Comment:
**Question:** do you think that we can find a way to simplify this expression here? Since I personally don't recommend that we introduce a complicated expression like this

Reply:
Fixed